src/socket-to-conn.ts

```diff
@@ -0,0 +1,89 @@
+import { abortableSource } from 'abortable-iterator'
+import { logger } from '@libp2p/logger'
+import type { MultiaddrConnection } from '@libp2p/interface-connection'
+import type { WebRTCPeer } from '@libp2p/webrtc-peer'
+import type { AbortOptions } from '@libp2p/interfaces'
+import type { Multiaddr } from '@multiformats/multiaddr'
+
+const log = logger('libp2p:webrtc-star:socket')
+
+export interface ToMultiaddrConnectionOptions extends AbortOptions {
+  remoteAddr: Multiaddr
+}
+
+/**
+ * Convert a socket into a MultiaddrConnection
+ * https://github.com/libp2p/js-libp2p-interfaces/tree/master/src/transport#multiaddrconnection
+ */
+export function toMultiaddrConnection (socket: WebRTCPeer, options: ToMultiaddrConnectionOptions): MultiaddrConnection {
```

Comment:
I don't think we want to expose a `MultiaddrConnection`. This will need to use the default upgrader which would cause nested muxing and encryption (noise over DTLS). This will make this implementation incompatible with the spec and the Rust and Go implementations. We should instead directly implement the `Connection` interface.